sylt-compiler/src/typechecker.rs

```diff
@@ -439,7 +453,15 @@ impl TypeChecker {
                         Some(var) => *var,
                         // NOTE(ed): This disallowes type-variables that are only used for
                         // constraints.
-                        None => return err_type_error!(self, span, ctx, todo_error!()),
+                        None => {
+                            return err_type_error!(
+                                self,
+                                span,
+                                ctx,
+                                TypeError::UnresolvedName(var.clone()),
+                                "Unused type-variable (only usages in the function signature are counted)"
```

Comment:
```suggestion
                                "Unused type-variable. (Only usages in the function signature are counted)"
```

Reply:
but i'm not sure how to handle multiple sentences when we don't usually include a period.

Reply:
maybe we should be able to write multiple lines? and let the error reporting handle indentation.

so we could do
```
Unusued type variable
(Only usages in the function signature are counted)
```

Reply:
A great PR - feel free to do it!